### DISCUSSION TITLE
How to sort pages with a custom criteria for date/time?

### DISCUSSION THREAD

**Discussion by PetersOtto** [datenstrom/community#852] [Ask a question] [answered]
Hello!

I am currently developing an event page. 
I sort the gigs by days and times.
At the moment it is structured cronologically. 

An example:
Gig 1 -> Day: 07.04.23 Time: 19:00
Gig 2 -> Day: 08.04.23 Time: 19:00
Gig 3 -> Day: 08.04.23 Time: 20:00
Gig 4 -> Day: 08.04.23 Time: 21:00

The best bands will play in the evening. Therefore it would be interesting to reverse the chronological »Time« sorting.

An example:
Gig 1 -> Day: 07.04.23 Time: 19:00
Gig 2 -> Day: 08.04.23 Time: 21:00
Gig 3 -> Day: 08.04.23 Time: 20:00
Gig 4 -> Day: 08.04.23 Time: 19:00

At the moment my code looks like this:

Header Blog Post:

```
---
Title: Blog example
Published: 2020-04-07
Author: Datenstrom
Layout: blog
GigDay: Sonntag
GigTiming: 08.04.23 20:00
---
```

start-blog.html:

```
....  (code before) ...

<?php foreach ($this->yellow->page->getPages("blog")->sort("GigTiming", true) as $page): ?>
<?php $gigTiming = $page->get("GigTiming"); $gigTimings = explode(" ", $gigTiming);?>

....  (default yellow code ) ...

<div class="post-container <?php echo $page->getHtml("entryClass") ?>">
            <a href="<?php echo $page->getLocation(true) ?>">
                <div class="post-gig">
                    <div class="post-day"> <p><?php echo $page->get("GigDay") ?></p></div>
                    <div class="post-date"> <p><?php echo $gigTimings[0] ?></p></div>
                    <div class="post-time"><p><?php echo $gigTimings[1] ?> Uhr</p></div>

....  (code after) ...
```

Has anyone done this before or perhaps an idea?

**Comment by schulle4u** [comment]
I don't have an idea for the actual task, but I remember that an [events diary extension](https://github.com/GiovanniSalmeri/yellow-diary) made by @GiovanniSalmeri already exists. Maybe this can give you a different approach for your project.

**Reply by PetersOtto** [reply]
Oh, great. I will have a look at it. Thanks

**Comment by markseuffert** [comment, marked as answer]
Here's a tip how sorting works in the [API for developers](https://datenstrom.se/yellow/help/api-for-developers). 

You can sort a collection of pages by any page setting. For example `pages->sort("modified")` or `pages->sort("title")`. If you cannot use one of your existing page settings, then you have to calculate a "sortScore" for each page. The tricky part is finding an algorithm that calculates a numerical value suitable for your criteria. Not sure how much work this would be for a developer, it requires some understanding of date formats. In any case the API has everything you need in terms of data storage/exchange. The easy part is the sorting. After you assigned a sort score to each page, you can sort them with `pages->sort("sortScore")`. If you want to see a practical code example, the search extensions works according to the same principle. See also datenstrom/community#748.

By the way, I would use the recommended date formats YYYY-MM-DD or YYYY-MM-DD HH:MM:SS. It makes working with dates so much easier. You can convert these formats to seconds with `strtotime()` or display them as a language specific date with `page->getDateHtml()`. See also datenstrom/community#273.

**Reply by PetersOtto** [reply]
Hello
Thank you very much for the detailed answer.
I will need a lot of time to work my way into it.

The hint to the recommended date formats has helped me a lot!
In some places my code looks much better now. &#128580;
